client/cli/lib/start.ts

```diff
@@ -1,33 +1,77 @@
-import * as liveServer from 'live-server';
-import bodyParser from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+import { resolve } from 'path';
 
 import build, { Options } from './build';
-import { validateDependencies } from './utils';
+import { getComponentDirectories, validateDependencies } from './utils';
 
 export const DEFAULT_PORT = 8383;
 
-export default async ({ port = DEFAULT_PORT }, buildOptions?: Options): Promise<any> => {
-  const { main, peerDependencies } = await build(buildOptions);
-
-  return liveServer.start({
-    port,
-    root: './dist',
-    file: main,
-    open: false,
-    cors: true,
-    middleware: [
-      bodyParser.json(),
-      (req, res, next) => {
-        if (req.method == 'GET') {
-          res.setHeader('Access-Control-Expose-Headers', 'ETag');
-        }
-
-        if (req.method === 'POST') {
-          validateDependencies(req.body, peerDependencies);
-        }
-
-        next();
+export type StartOptions = {
+  port?: number;
+  workspaces?: string[];
+};
+
+export default async (
+  logger,
+  { port = DEFAULT_PORT, workspaces }: StartOptions,
+  buildOptions?: Options
+): Promise<any> => {
+  const dir = buildOptions && buildOptions.dir;
+
+  const dirs = getComponentDirectories(dir, workspaces);
+  const builds = await Promise.all(
+    dirs.map(async dir => {
+      const result = await build({ ...buildOptions, dir });
+      return { ...result, dir };
+    })
+  );
+
+  const index: Record<string, any> = builds.reduce(
+    (acc, { name, ...result }) => ({
+      ...acc,
+      [name]: result
+    }),
+    {}
+  );
+
+  const app = express();
+
+  app.set('etag', 'strong');
+
+  app.use(express.json(), cors());
+
+  app
+    .route('/:component')
+    .all((req, res, next) => {
+      if (!(req.params.component in index)) {
+        logger.warn(`component "${req.params.component}" was not found`);
+        return res.sendStatus(404);
       }
-    ]
+      req.params.component = index[req.params.component];
+      next();
+    })
+    .get((req, res) => {
+      res.setHeader('Access-Control-Expose-Headers', 'ETag');
+      const component = req.params.component;
+      res.sendFile(resolve(process.cwd(), component.dir, 'dist', component.main));
+    })
+    .post((req, res) => {
+      const component = req.params.component;
+      validateDependencies(req.body, component.peerDependencies);
+      res.sendStatus(200);
+    })
+    .all((req, res) => {
+      return res.sendStatus(404);
+    });
+
+  app.post('/host/register', (req, res) => {
+    res.sendStatus(200);
```

Comment:
`validateDependencies` here

Reply:
which peerDependencies am I supposed to use? we have a lot of packages